tv_apps/browser/js/sync/settings.js

```diff
@@ -261,6 +295,13 @@
         navigator.mozId.watch({
           wantIssuer: 'firefox-accounts',
           onlogin: () => {
+            // This callback is fired everytime a new account logs in,
+            // but we only want to enable the history setting when this
+            // callback is called because the refresh auth flow succeeded.
```

Comment:
Also, from the code it seems `onhistorychecked` is called both when `this.elements.collectionHistory.checked` changes to `true`, and when it changes to `false`. But it seems the checkbox is always unchecked after a `navigator.mozId` error, here: https://github.com/mozilla-b2g/gaia/pull/32779/files#diff-50ac43c342df1ca3d63486ce5b2a1057R313, isn't that also an asymmetry?